src/CmsBundle/DependencyInjection/OpiferCmsExtension.php

```diff
@@ -68,6 +68,10 @@ public function getParameters(array $config)
             'opifer_cms.allowed_locales' => ['en'],
         ];
 
+        foreach($config['permissions'] as $role => $permissions) {
+            $params['opifer_cms.permissions.'.strtolower($role)] = $permissions;
```

Comment:
I'm thinking if it shouldn't be better to simply pass the whole permission mapping onto the `opifer_cms.permissions` parameter, instead of creating a new parameter for each role.

This way you can also avoid injecting the whole `ServiceContainer` into the `ActionVoter`, and just inject the `PermissionRegistry` and the `opifer_cms.permissions` parameter.